src/main/java/com/googlecode/lanterna/graphics/AbstractTextGraphics.java

```diff
@@ -344,4 +357,31 @@ public TextGraphics newTextGraphics(TerminalPosition topLeftCorner, TerminalSize
     private TextCharacter newTextCharacter(char character) {
         return new TextCharacter(character, foregroundColor, backgroundColor, activeModifiers);
     }
+
+    private EnumSet<SGR> updateLocalModifiers(String controlSequence, EnumSet<SGR> localModifiers) {
+        char controlCodeType = controlSequence.charAt(controlSequence.length() - 1);
+        controlSequence = controlSequence.substring(0, controlSequence.length() - 1);
```

Comment:
Why not just controlSequence.substring(**2**, controlSequence.length() - 1) and avoid this rather expensive two-steps elimination of Esc and `[` ?